### PR TITLE
use centralized builder image

### PR DESCRIPTION
**Pull request by rajrohanyadav**
Use the centralized docker image for building.

## How is it tested?
Created a pre-release and the action trigged successfully to build and push the artifacts to s3 in the correct location.

https://github.com/newrelic/nri-docker/actions/runs/11907011686

image

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
+VALIDATE_DEPS    	= golang.org/x/lint/golint
+INTEGRATIONS_DIR 	= /var/db/newrelic-infra/newrelic-integrations/
+CONFIG_DIR       	= /etc/newrelic-infra/integrations.d
+GO_FILES        	:= ./src/
+GOOS             	= GOOS=linux
+GO               	= $(GOOS) go
+GO_VERSION 			?= $(shell grep '^go ' go.mod | awk '{print $$2}')
+UBUNTU_VERSION 		?= "16.04"
+BUILDER_IMAGE_TAG 	?= "latest"
+BUILDER_IMAGE 		?= "ghcr.io/newrelic/coreint-automation:latest-go$(GO_VERSION)-ubuntu$(UBUNTU_VERSION)"
```
Comment (paologallinaharbur) [inline Makefile:13]: Oh I think I misunderstood how we were using it, so we leverage pre-compiled images, we do not trigger the creation each time

File: Makefile [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
+SRC_DIR          	= ./src/
+VALIDATE_DEPS    	= golang.org/x/lint/golint
+INTEGRATIONS_DIR 	= /var/db/newrelic-infra/newrelic-integrations/
+CONFIG_DIR       	= /etc/newrelic-infra/integrations.d
+GO_FILES        	:= ./src/
+GOOS             	= GOOS=linux
+GO               	= $(GOOS) go
+GO_VERSION 			?= $(shell grep '^go ' go.mod | awk '{print $$2}')
+UBUNTU_VERSION 		?= "16.04"
+BUILDER_IMAGE_TAG 	?= "latest"
```
Comment (paologallinaharbur) [inline Makefile:12]: if no one is setting this I would not make it configurable and just place them in the BUILDER_IMAGE (that is already configurable)

File: Makefile (hunk 1 above)
Comment (rajrohanyadav) [reply Makefile:13]: Yes, we only trigger when we want a new image built for a new version, OR if some changes are pushed to main.

File: build/ci.mk [hunk 3]
```diff
@@ -17,23 +18,23 @@ ci/debug-container: ci/deps
 			-e GPG_MAIL \
 			-e GPG_PASSPHRASE \
 			-e GPG_PRIVATE_KEY_BASE64 \
-			$(BUILDER_TAG) bash
+			$(BUILDER_IMAGE) bash
```
Comment (paologallinaharbur) [inline build/ci.mk:21]: why are we passing it here?

File: build/ci.mk [hunk 4]
```diff
@@ -17,23 +18,23 @@ ci/debug-container: ci/deps
...
 
 .PHONY : ci/validate
 ci/validate: ci/deps
 	@docker run --rm -t \
 			--name "nri-$(INTEGRATION)-validate" \
 			-v $(CURDIR):/go/src/github.com/newrelic/nri-$(INTEGRATION) \
 			-w /go/src/github.com/newrelic/nri-$(INTEGRATION) \
-			$(BUILDER_TAG) make validate
+			$(BUILDER_IMAGE) make validate
```
Comment (paologallinaharbur) [inline build/ci.mk:29]: who is setting and then reading this environment variable? It comes from the makefile, and it is passed to a second makefile?

File: Makefile (hunk 2 above)
Comment (rajrohanyadav) [reply Makefile]: Makes sense. Done. Thanks for suggesting.

File: build/ci.mk (hunk 4 above)
Comment (paologallinaharbur) [reply build/ci.mk:29]: ah ok I got it, it is replacing BUILDER_TAG

**Review by paologallinaharbur — APPROVED** [review]

**Review by alvarocabanas — APPROVED** [review]
LGTM, nice work

**Pull request merged**